src/Messages.php

```diff
@@ -72,9 +72,16 @@ public function addMessage($key, $message)
         if (!isset($this->storage[$this->storageKey][$key])) {
             $this->storage[$this->storageKey][$key] = array();
         }
-        
+
         //Push onto the array
-        $this->storage[$this->storageKey][$key][] = (string)$message;
+        if (
+            is_array($message) === true
+            || is_object($message) === true
+        ) {
```

Comment:
@NigelGreenway yeah I think I would leave that down to the developer to decide. I now understand why we were only storing string as the session store might not know how to deal with objects, but I also think if a developer decides to throw an object into the session, they should know what their store can handle or deal with how it should handle what they throw at it.

My expectation was to see this:

```php
$this->storage[$this->storageKey][$key][] = (string)$message;
```

become:

```php
$this->storage[$this->storageKey][$key][] = $message;
```

Here I would be making no assumption as to what the developer want to store, `$message` could be a string, object or array if they wanted. We might however need to note this in the README.